### PR TITLE
Don't update pre-releases

### PR DESCRIPTION
**Pull request by grdowns**
Do not try to update if the current version is greater than the latest release or incomparable

### CODE REVIEW

File: Extension/src/githubAPI.ts [hunk 1]
```diff
@@ -130,7 +130,17 @@ export interface BuildInfo {
  */
 export async function getTargetBuildInfo(updateChannel: string): Promise<BuildInfo> {
     return getReleaseJson()
-        .then(builds => getTargetBuild(builds, updateChannel))
+        .then(builds => {
+            // If the user version is greater than or incomparable to the latest available verion then there is no need to update
+            // Allows testing pre-releases without accidentally downgrading to the latest version
+            const userVersion: PackageVersion = new PackageVersion(util.packageJson.version);
+            const latestVersion: PackageVersion = new PackageVersion(builds[0].name);
+            if (userVersion.isGreaterThan(latestVersion) || (userVersion.suffix && userVersion.suffix !== 'insiders')) {
```
Comment (pieandcakes) [inline Extension/src/githubAPI.ts:142]: might want to do a starts with 'insiders' or contains 'insiders' since we have the insiders2 problem right now.

**Review by pieandcakes — DISMISSED** [review]

File: Extension/src/githubAPI.ts (hunk 1 above)
Comment (bobbrow) [reply Extension/src/githubAPI.ts:142]: for insiders2, the 2 will be removed and `.suffix` will just have `"insiders"` and `.suffixVersion` will have 2

Comment (bobbrow) [inline Extension/src/githubAPI.ts:142]: You don't need to check `userVersion.suffix` before checking `userVersion.suffix !== 'insiders'`

File: Extension/src/githubAPI.ts [hunk 2]
```diff
@@ -130,7 +130,17 @@ export interface BuildInfo {
  */
 export async function getTargetBuildInfo(updateChannel: string): Promise<BuildInfo> {
     return getReleaseJson()
-        .then(builds => getTargetBuild(builds, updateChannel))
+        .then(builds => {
+            // If the user version is greater than or incomparable to the latest available verion then there is no need to update
+            // Allows testing pre-releases without accidentally downgrading to the latest version
+            const userVersion: PackageVersion = new PackageVersion(util.packageJson.version);
+            const latestVersion: PackageVersion = new PackageVersion(builds[0].name);
```
Comment (bobbrow) [inline Extension/src/githubAPI.ts:141]: is it safe to just pass in `build[0]` here?

**Review by bobbrow — CHANGES_REQUESTED** [review]

File: Extension/src/githubAPI.ts (hunk 1 above)
Comment (grdowns) [reply Extension/src/githubAPI.ts:142]: I believe that would change the behavior to make it such that users on a release version would never be able to update. Is that right?

Comment (grdowns) [reply Extension/src/githubAPI.ts:142]: I can test this

Comment (grdowns) [reply Extension/src/githubAPI.ts:142]: Yeah, that's the case. I'll leave this in

Comment (bobbrow) [reply Extension/src/githubAPI.ts:142]: Sorry, my original comment was bogus.

**Review by bobbrow — APPROVED** [review]

**Pull request merged**